src/caffe/layer_factory.cpp

```diff
@@ -67,6 +67,7 @@ shared_ptr<Layer<Dtype> > GetConvolutionLayer(
 #endif
   } else {
     LOG(FATAL) << "Layer " << param.name() << " has unknown engine.";
+    throw;  // Avoids missing return warning
```

Comment:
`return NULL` is more clear than `throw` at least in my reading, and consistent with #3362.